core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -557,7 +563,7 @@ public static void updateDatabase(Connection dbConnection, String sqlStmt, Objec
             }
             throw new UserStoreException(errorMessage, e);
         } finally {
-            DatabaseUtil.closeAllConnections(null, prepStmt);
+            close(dbConnection);
```

Comment:
Since the connection getting passed as a parameter to this method, it shouldn't close the connection inside this method.

Need to close the connection in the original place where it originally (create and )close the dbConnection.